src/uint_traits.rs

```diff
@@ -0,0 +1,86 @@
+#![allow(missing_docs)]
+
+use crypto_bigint::{BoxedUint, Integer, Random, RandomMod, Uint};
+use rand_core::CryptoRngCore;
+
+// would be nice to have: *Assign traits; arithmetic traits for &self (BitAnd and Shr in particular);
+pub trait UintLike: Integer + RandomMod {
+    fn bit_vartime(&self, index: u32) -> bool;
+    fn trailing_zeros(&self) -> u32;
+    fn trailing_ones(&self) -> u32;
+    fn sqrt_vartime(&self) -> Self;
+    fn shr_vartime(&self, shift: u32) -> Self;
+    fn shl_vartime(&self, shift: u32) -> Self;
+    fn random_bits(rng: &mut impl CryptoRngCore, bit_length: u32) -> Self;
+    fn try_into_u32(&self) -> Option<u32> {
+        self.as_ref()[0].0.try_into().ok()
+    }
+}
+
+// Uint<L> impls
+
+impl<const L: usize> UintLike for Uint<L> {
+    fn trailing_zeros(&self) -> u32 {
+        Self::trailing_zeros(self)
+    }
+
+    fn trailing_ones(&self) -> u32 {
+        Self::trailing_ones(self)
+    }
+
+    fn bit_vartime(&self, index: u32) -> bool {
+        Self::bit_vartime(self, index)
+    }
+
+    fn sqrt_vartime(&self) -> Self {
+        Self::sqrt_vartime(self)
+    }
+
+    fn shr_vartime(&self, shift: u32) -> Self {
+        Self::wrapping_shr_vartime(self, shift)
+    }
+
+    fn shl_vartime(&self, shift: u32) -> Self {
+        Self::wrapping_shl_vartime(self, shift)
+    }
+
+    fn random_bits(rng: &mut impl CryptoRngCore, bit_length: u32) -> Self {
+        if bit_length > Self::BITS {
+            panic!("The requested bit length ({bit_length}) is larger than the chosen Uint size");
+        }
+        let random = Self::random(rng);
+        random >> (Self::BITS - bit_length)
+    }
+}
+
+impl UintLike for BoxedUint {
+    fn trailing_zeros(&self) -> u32 {
+        Self::trailing_zeros(self)
+    }
+
+    fn trailing_ones(&self) -> u32 {
+        Self::trailing_ones(self)
+    }
+
+    fn bit_vartime(&self, index: u32) -> bool {
+        Self::bit_vartime(self, index)
+    }
+
+    fn sqrt_vartime(&self) -> Self {
+        Self::sqrt_vartime(self)
+    }
+
+    fn shr_vartime(&self, shift: u32) -> Self {
+        Self::wrapping_shr_vartime(self, shift)
+    }
+
+    fn shl_vartime(&self, shift: u32) -> Self {
+        Self::wrapping_shl_vartime(self, shift)
+    }
+
+    fn random_bits(rng: &mut impl CryptoRngCore, bit_length: u32) -> Self {
```

Comment:
I am sure @fjarri is already aware but this implementation doesn't do what it says to do. In this implementation `random.bits_precision()` returns `bit_length`, so `random >> (bits_precision - bit_length)` will always return `0`.

Reply:
I actually was not aware of it since I only ran tests with `Uint`s. Thanks!